### PR TITLE
Use farming mod for hoes

### LINKED ISSUE #29: Disable hoes
Currently ```moreores``` has its own hoe implementation which does not check whether the ```farming``` mod is installed. This makes it possible to craft senseless mithril and silver hoes, which are unusable, because the ```farming:soil``` node is undefined.

The bronze, mese and diamond hoes have been deprecated by https://github.com/minetest/minetest_game/commit/9c459e77ac943f10c74de3bdd420e07a2b8bb56f

The mithril and silver hoe should also be deprecated in ```moreores```.

### PR DESCRIPTION
**Pull request by SpaghettiToastBook**
Also adds a deprecation mechanism in case the it's decided that the silver and/or mithril hoes should be deprecated in the same manner as the bronze, mese, and diamond ones.

Fixes #29.

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -253,7 +241,8 @@ local oredefs = {
 				cracky = {times = {[1] = 2.60, [2] = 1.00, [3] = 0.60}, uses = 100, maxlevel= 1}
 			},
 			hoe = {
-				uses = 300
+				uses = 300,
+				deprecated = false,
```
Comment (SmallJoker) [inline init.lua]: Why? It defaults to `false` anyway?

File: init.lua [hunk 2]
```diff
@@ -285,7 +274,8 @@ local oredefs = {
 				cracky = {times = {[1] = 2.25, [2] = 0.55, [3] = 0.35}, uses = 200, maxlevel= 2}
 			},
 			hoe = {
-				uses = 1000
+				uses = 1000,
+				deprecated = false,
```
Comment (SmallJoker) [inline init.lua]: Also here

File: init.lua [hunk 3]
```diff
@@ -188,50 +157,69 @@ local function add_ore(modname, description, mineral_name, oredef)
...
-		minetest.register_tool(fulltool_name, tdef)
-		minetest.register_alias(tool_name .. tool_post, fulltool_name)
-		if oredef.makes.ingot then
-			minetest.register_craft({
-				output = fulltool_name,
-				recipe = get_recipe(ingot, tool_name)
-			})
-		end
+        
+        if tool_name == "hoe" and minetest.get_modpath("farming") then
```
Comment (SmallJoker) [inline init.lua]: Please use tabs to indent.

Comment (SpaghettiToastBook) [reply init.lua]: Okay.

File: init.lua (hunk 1 above)
Comment (SpaghettiToastBook) [reply init.lua]: Just an oversight on my part.

**Comment by SpaghettiToastBook** [comment]
Also fixes #21.

**Comment by Calinou** [comment]
Thanks for your contribution :smiley:

This is superseded by a [simplified version](https://github.com/minetest-mods/moreores/commit/4db884e00cf87f734146a098732b70a1f51934a3) based on your pull request, so I'm closing this.

**Pull request closed without merging**